cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/metric/ContainerMetricUtils.java

```diff
@@ -0,0 +1,148 @@
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public final class ContainerMetricUtils {
+  // Paths used to get cgroup information
+  private static final String QUOTA_PATH = "/sys/fs/cgroup/cpu/cpu.cfs_quota_us";
+  private static final String PERIOD_PATH = "/sys/fs/cgroup/cpu/cpu.cfs_period_us";
+  // Unix command to execute inside a Linux container to get the number of logical processors available to the node
+  private static final String NPROC = "nproc";
+  // A CPU quota value of -1 indicates that the cgroup does not adhere to any CPU time restrictions
+  public static final int NO_CPU_QUOTA = -1;
+
+  private ContainerMetricUtils() { };
+
+  /**
+   * Reads Cgroups CPU period from Cgroups file. Value has a lowerbound of 1 millisecond and  an upperbound of 1 second
+   * according to https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt
+   *
+   * @return Cgroups CPU period in microseconds as a double.
+   */
+  private static double getCpuPeriod() {
+    return Double.parseDouble(readFile(CgroupFiles.PERIOD_PATH.getValue()));
+  }
+
+  /**
+   * Reads Cgroups CPU quota from Cgroups file. The value has lowerbound of 1 millisecond
+   * according to https://www.kernel.org/doc/Documentation/scheduler/sched-bwc.txt
+   *
+   * @return Cgroups CPU quota in microseconds as a double.
+   */
+  private static double getCpuQuota() {
+    return Double.parseDouble(readFile(CgroupFiles.QUOTA_PATH.getValue()));
+  }
+
+  /**
+   * Gets the the number of logical cores available to the node.
+   * <p>
+   * We can get this value while running in a container by using the "nproc" command.
+   * Using other methods like OperatingSystemMXBean.getAvailableProcessors() and
+   * Runtime.getRuntime().availableProcessors() would require disabling container
+   * support (-XX:-UseContainerSupport) since these methods are aware of container
+   * boundaries
+   *
+   * @return Number of logical processors on node
+   */
+  private static int getAvailableProcessors() {
+    int proc = 1;
+    try {
+      InputStream in = Runtime.getRuntime().exec(NPROC).getInputStream();
+      proc = Integer.parseInt(readInputStream(in));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return proc;
+  }
+
+  private static String readFile(String path) {
+    try {
+      return readInputStream(new FileInputStream(path));
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  private static String readInputStream(InputStream in) {
+    try {
+      BufferedReader br = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+      String stream = br.readLine();
+      if (stream != null) {
+        return stream;
+      } else {
+        throw new EmptyInputStreamException("Nothing was read from stream " + in);
```

Comment:
Should we avoid introducing a new exception type by throwing an `IllegalArgumentException` here?